plugins/scaffolder/src/components/fields/EntityPicker/EntityPicker.tsx

```diff
@@ -32,17 +32,33 @@ import FormControl from '@material-ui/core/FormControl';
 import Autocomplete, {
   AutocompleteChangeReason,
 } from '@material-ui/lab/Autocomplete';
-import React, { useCallback, useEffect } from 'react';
+import React, { useCallback, useEffect, useMemo } from 'react';
 import useAsync from 'react-use/lib/useAsync';
 import {
+  EntityPickerFilterQuery,
   EntityPickerFilterQueryValue,
   EntityPickerProps,
   EntityPickerUiOptions,
-  EntityPickerFilterQuery,
 } from './schema';
 
 export { EntityPickerSchema } from './schema';
 
+/**
+ * @param string the string to interpolate
+ * @param params values to used as replacements
+ * @see https://stackoverflow.com/a/41015840/1666829
+ */
+const interpolate = (string: string, params: object): string => {
+  const names = Object.keys(params);
+  const vals = Object.values(params);
+  try {
+    // eslint-disable-next-line no-new-func
+    return new Function(...names, `return \`${string}\`;`)(...vals);
```

Comment:
We go to quite some lengths to avoid eval and the function constructor, since they are huge security risks and are also treated as such by default by CSP in browsers which adds additional complexity for one's setup. Unfortunately I'd personally vote against this PR for this line of code alone.

We do add `script-src: 'unsafe-eval'` at this point because of `ajv` needing it in non-compiled mode, but that's the single reason and we've been very passionately wanted to get rid of that one final reason to keep that rule around. We just haven't had time to address that yet.

Reply:
No Problem, I can use regexp to make the replacements as well.

Reply:
Are you tracking removal of `unsafe-eval` anywhere? Would love to have an issue to subscribe to